pkg/controllers/managementuser/snapshotbackpopulate/snapshotbackpopulate.go

```diff
@@ -77,58 +89,175 @@ func (h *handler) OnChange(key string, configMap *corev1.ConfigMap) (*corev1.Con
+			if existingSnapshotCR, err = h.etcdSnapshots.UpdateStatus(existingSnapshotCR); err != nil && !apierrors.IsNotFound(err) {
+				return configMap, fmt.Errorf("rkecluster %s/%s: error while setting status missing=true on etcd snapshot %s/%s: %w", cluster.Namespace, cluster.Name, existingSnapshotCR.Namespace, existingSnapshotCR.Name, err)
+			}
+			continue
+		}
+		currentEtcdSnapshotsToKeep[existingSnapshotCR.Name] = existingSnapshotCR
+	}
+
+	// iterate over the actual etcd snapshots that are in the management cluster
+	// if the snapshot is found in the desired etcd snapshots, check to see if an update needs to be made
+	// otherwise, create the etcd snapshot CR
+	for _, cmGeneratedSnapshot := range actualEtcdSnapshots {
+		if snapshot, ok := currentEtcdSnapshotsToKeep[cmGeneratedSnapshot.Name]; ok {
+			if !equality.Semantic.DeepEqual(cmGeneratedSnapshot.SnapshotFile, snapshot.SnapshotFile) || !equality.Semantic.DeepEqual(cmGeneratedSnapshot.Status, snapshot.Status) {
+				logrus.Debugf("[snapshotbackpopulate] rkecluster %s/%s: updating etcd snapshot %s/%s as it differed from the actual snapshot config map %v vs %v", cluster.Namespace, cluster.Name, snapshot.Namespace, snapshot.Name, cmGeneratedSnapshot.SnapshotFile, snapshot.SnapshotFile)
+				logrus.Tracef("[snapshotbackpopulate] rkecluster %s/%s: updating etcd snapshot %s/%s: %+v", cluster.Namespace, cluster.Name, cmGeneratedSnapshot.Namespace, cmGeneratedSnapshot.Name, cmGeneratedSnapshot)
+				snapshot = snapshot.DeepCopy()
+				// keep a copy of the metadata and message
+				md := snapshot.SnapshotFile.Metadata
+				msg := snapshot.SnapshotFile.Message
+				snapshot.SnapshotFile = cmGeneratedSnapshot.SnapshotFile
+				// restore the metadata and message as those may have been lost
+				snapshot.SnapshotFile.Metadata = md
+				snapshot.SnapshotFile.Message = msg
+				snapshot, err = h.etcdSnapshots.Update(snapshot)
+				if err != nil {
+					return configMap, fmt.Errorf("rkecluster %s/%s: error while updating etcd snapshot %s/%s: %w", cluster.Namespace, cluster.Name, snapshot.Namespace, snapshot.Name, err)
+				}
+				if snapshot.Status.Missing {
+					snapshot.Status.Missing = false
+					// the kube-apiserver only accepts status updates on deliberate subresource status updates which is why we have to double-call an update here if the missing is set incorrectly
+					if _, err := h.etcdSnapshots.UpdateStatus(snapshot); err != nil && !apierrors.IsNotFound(err) {
+						return configMap, fmt.Errorf("rkecluster %s/%s: error while setting status missing=false on etcd snapshot %s/%s: %w", cluster.Namespace, cluster.Name, snapshot.Namespace, snapshot.Name, err)
+					}
+				}
+			}
+		} else {
+			// create the snapshot in the mgmt cluster
+			logrus.Debugf("[snapshotbackpopulate] rkecluster %s/%s: creating etcd snapshot %s/%s as it differed from the actual snapshot config map", cluster.Namespace, cluster.Name, cmGeneratedSnapshot.Namespace, cmGeneratedSnapshot.Name)
```

Comment:
Not sure if `as it differed from the actual snapshot config map` makes sense here as we don't have this snapshot object at all yet, so there is nothing to "differ". Remove this wording?

Reply:
This is a debug message anyway, we can correct this later on but I don't want to address this at this point.